gronx.go

```diff
@@ -96,11 +96,44 @@ func Segments(expr string) ([]string, error) {
 // SegmentsDue checks if all cron parts are due.
 // It returns bool. You should use IsDue(expr) instead.
 func (g *Gronx) SegmentsDue(segs []string) (bool, error) {
-	for pos, seg := range segs {
+	skipMonthDayCheck := false
+	for i := 0; i < len(segs); i++ {
+		pos := len(segs) - 1 - i
+		seg := segs[pos]
+		isMonthDay, isWeekday := pos == 3, pos == 5
+
 		if seg == "*" || seg == "?" {
 			continue
 		}
 
+		if isMonthDay && skipMonthDayCheck {
+			continue
+		}
+
+		if isWeekday {
+			segIsIntersecting := strings.Index(seg, "*/") == 0
+			monthDaySeg := segs[3]
+			monthDaySegIsIntersecting := strings.Index(monthDaySeg, "*") == 0 || monthDaySeg == "?"
+			intersectCase := segIsIntersecting || monthDaySegIsIntersecting
+
+			if !intersectCase {
```

Comment:
i could not understand the behavior of intserction/union of dom/dow.

Reply:
If either day of week or day of month segments starts with a star, then we are in interection case, meaning both conditions must be satisfied. If not, it is a union case, so either must be satisfied.

So here `!intersectCase` means we are in union case.
The iterations always check weekday first. Here at this line, if we are in union, we check if weekday segment is due or if monthday is due. If either is due we can continue onto the next segment. The skip boolean is just here to not check monthday again since we already do it here at the day of week position.